CONTRIBUTING.md

```diff
@@ -0,0 +1,100 @@
+# Contributing to SleepECG
+If you want to implement a new feature, fix an existing bug or help improve SleepECG in any other way (such as adding or improving documentation), please consider submitting a [pull request](https://github.com/cbrnr/sleepecg/pulls) on GitHub. It might be a good idea to open an [issue](https://github.com/cbrnr/sleepecg/issues) beforehand and discuss your planned contributions with the developers.
+
+Before you start working on your contribution, please make sure to follow the guidelines described in this document.
+
+
+## GitHub workflow
+### Setup
+- Create a [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) of the [GitHub repository](https://github.com/cbrnr/sleepecg).
+- Clone the fork to your machine: `git clone https://github.com/<your-username>/sleepecg`.
+- Make sure your [username](https://docs.github.com/en/get-started/getting-started-with-git/setting-your-username-in-git) and [email](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git) are configured to match your GitHub account.
+- Add the original repository (also called _upstream_) as a remote to your local clone: `git remote add upstream git@github.com:cbrnr/sleepecg.git`.
+
+
+### Add a feature or fix a bug
+- Create and switch to a new branch: `git switch -c branch-name`
+    - `branch-name` should be representative of what you are working on.
+- Make changes and commit them.
+- Push the changes to your remote fork.
+- Create a [pull request (PR)](https://github.com/cbrnr/sleepecg/pulls).
+- Add an entry to `CHANGELOG.md` (section 'UNRELEASED') where you mention the corresponding PR and (if you desire) your name.
+
+
+### Rebasing
+If another PR is merged while you are working on something, a merge conflict may arise. To resolve it, perform the following steps in your local clone:
+- Switch to the main branch: `git switch main`.
+- Pull the current changes: `git pull upstream main`.
+- Switch back to the branch you are working on: `git switch branch-name`.
+- Rebase your commits onto main: `git rebase main`.
+- Push to your remote fork: `git push` (might require `git push --force`).
+
+
+## Development enviroment
+Make sure to use Python 3.7. You might want to [create a virtual environment](https://docs.python.org/3/library/venv.html#creating-virtual-environments) instead of installing everything into your main environment. In the root of the local clone of your fork, run
```

Comment:
```suggestion
Make sure to use at least Python 3.7. You might want to [create a virtual environment](https://docs.python.org/3/library/venv.html#creating-virtual-environments) instead of installing everything into your main environment. In the root of the local clone of your fork, run
```

Reply:
This might lead to contributors using features incompatible with 3.7, which will then fail during CI,

Reply:
True!